src/abstractdataframe/join.jl

```diff
@@ -430,23 +345,622 @@ function Base.join(df1::AbstractDataFrame, df2::AbstractDataFrame;
     return joined
 end
 
-function Base.join(df1::AbstractDataFrame, df2::AbstractDataFrame,
-                   dfs::AbstractDataFrame...;
-                   on::Union{Symbol, AbstractVector{Symbol}} = Symbol[],
-                   kind::Symbol = :inner, makeunique::Bool=false,
-                   validate::Union{Pair{Bool, Bool}, Tuple{Bool, Bool}}=(false, false))
-    if !(kind in (:inner, :outer, :cross))
-        throw(ArgumentError("Only inner, outer, and cross joins are supported when " *
-                            "joining more than two data frames"))
-    end
-    join(join(df1, df2, on=on, kind=kind, makeunique=makeunique, validate=validate),
-         dfs..., on=on, kind=kind, makeunique=makeunique, validate=validate)
-end
+"""
+    innerjoin(df1, df2; on = Symbol[], makeunique = false,
+              indicator = nothing, validate = (false, false))
+    innerjoin(df1, df2, dfs...; on = Symbol[], makeunique = false,
+              validate = (false, false))
+
+Inner join two or more data frame objects and return a `DataFrame` containing
+the result.
+
+# Arguments
+- `df1`, `df2`, `dfs...`: the `AbstractDataFrames` to be joined
+
+# Keyword Arguments
+- `on` : A column name to join `df1` and `df2` on. If the columns on which
+  `df1` and `df2` will be joined have different names, then a `left=>right`
+  pair can be passed. It is also allowed to perform a join on multiple columns,
+  in which case a vector of column names or column name pairs can be passed
+  (mixing names and pairs is allowed). If more than two data frames are joined
+  then only a column name or a vector of column names are allowed.
+  `on` is a required argument.
+- `makeunique` : if `false` (the default), an error will be raised
+  if duplicate names are found in columns not joined on;
+  if `true`, duplicate names will be suffixed with `_i`
+  (`i` starting at 1 for the first duplicate).
+- `indicator` : Default: `nothing`. If a `Symbol`, adds categorical indicator
```

Comment:
This doesn't make a lot of sense here, right (since rows must appear in both inputs)? Maybe better remove it. For other functions, it might be worth updating the description to mention that result will never be `"left_only"`/`"right_only"`/`"both"`.

Reply:
Indeed. I have cleaned such issues in other places. Actually I will remove it here and also for `antijoin` and `semijoin` where it currently even errors.

It seems that the idea to split the functions was good, as it simplifies thinking about their contract (and things that slipped through can be caught).